yearclass/src/main/java/com/facebook/device/yearclass/DeviceInfo.java

```diff
@@ -50,6 +58,39 @@ public static int getNumberOfCPUCores() {
     return cores;
   }
 
+  /**
+   * Tries to read file contents from the file location to determine the number of cores on device.
+   * @param fileLocation The location of the file with CPU information
+   * @return Number of CPU cores in the phone, or DEVICEINFO_UKNOWN = -1 in the event of an error.
+   */
+  private static int getCoresFromFileInfo(String fileLocation) {
+    try {
+      InputStream is = new FileInputStream(fileLocation);
+      BufferedReader buf = new BufferedReader(new InputStreamReader(is));
+      String fileContents = buf.readLine();
+      return getCoresFromFileString(fileContents);
+    } catch (IOException e) {
+      return DEVICEINFO_UNKNOWN;
+    }
+  }
+
+  /**
+   * Converts from a CPU core information format to number of cores.
+   * @param str The CPU core information string, in the format of "0-N"
+   * @return The number of cores represented by this string
+   */
+  protected static int getCoresFromFileString(String str) {
```

Comment:
"protected" is something I'm not used to seeing on a static method. I think it should just be package scoped (i.e. no qualifier).